File: language-examples/csharp/rest/RestResetCorpus.cs
using System.Text.Json;

class RestResetCorpus
{
    /// <summary>
    /// Calls Vectara platform to reset a corpus.
    /// </summary>
    /// <param name="customerId"> The unique customer ID in Vectara platform. </param>
    /// <param name="corpusId"> The unique ID of the corpus to be reset. </param>
    /// <param name="jwtToken"> A valid authentication token. </param>
    /// <throws> Exception if Reset operation fails. </throws>
    public static void ResetCorpus(long customerId, long corpusId, string jwtToken)
    {
        using (var client = new HttpClient())
        {
            try
            {
                var request = new HttpRequestMessage
                {
                    RequestUri = new Uri($"https://{ServerEndpoints.commonEndpoint}/v1/reset-corpus"),
                    Method = HttpMethod.Post,
                };

                Dictionary<string, object> data = new()
                {
                    { "customer_id", customerId },
                    { "corpus_id", corpusId }
                };

                string jsonData = JsonSerializer.Serialize(data);
                Console.WriteLine(jsonData);

                request.Content = new StringContent(jsonData);
                request.Content.Headers.Remove("Content-Type");
                request.Content.Headers.Add("Content-Type", "application/json");

                request.Headers.Add("Authorization", $"Bearer {jwtToken}");
                request.Headers.Add("customer-id", customerId.ToString());

                HttpResponseMessage response = client.Send(request);
                string result = response.Content.ReadAsStringAsync().Result;

                Console.WriteLine(result);
            }
            catch (Exception)
            {
                throw;
            }
        }
    }
}

File: language-examples/python/vectara-rest/rest_reset_corpus.py
"""Simple example of using the Vectara REST API for resetting a corpus.
"""

import json
import logging
import requests

def _get_reset_corpus_json(customer_id: int, corpus_id: int):
    """ Returns a reset corpus json. """
    corpus = {}
    corpus["customer_id"] = customer_id
    corpus["corpus_id"] = corpus_id

    return json.dumps(corpus)

def reset_corpus(customer_id: int, corpus_id: int, admin_address: str, jwt_token: str):
    """Reset a corpus.
    Args:
        customer_id: Unique customer ID in vectara platform.
        corpus_id: Corpus ID in vectara platform.
        admin_address: Address of the admin server. e.g., api.vectara.io
        jwt_token: A valid Auth token.

    Returns:
        (response, True) in case of success and returns (error, False) in case of failure.
    """

    post_headers = {
        "customer-id": f"{customer_id}",
        "Authorization": f"Bearer {jwt_token}"
    }
    response = requests.post(
        f"https://{admin_address}/v1/reset-corpus",
        data=_get_reset_corpus_json(customer_id, corpus_id),
        verify=True,
        headers=post_headers)

    if response.status_code != 200:
        logging.error("Reset Corpus failed with code %d, reason %s, text %s",
                       response.status_code,
                       response.reason,
                       response.text)
        return response, False

    message = response.json()
    if message["status"]["code"] != "OK":
        logging.error("Delete Corpus failed with status: %s", message.status)
        return message.status, False

    return message, True
